pandas/util/_decorators.py

```diff
@@ -503,3 +503,24 @@ def indent(text: str | None, indents: int = 1) -> str:
     "future_version_msg",
     "Substitution",
 ]
+
+
+def set_module(module):
```

Comment:
Just came across this today, I do not see the benefit of having this decorator. We either have:

```python
class DataFrame:
    __module__ = "pandas"
```

or

```python
@set_module("pandas')
class DataFrame:
    ...
```

There is a cost, particularly for reading code you have to chase down the definition. I'm also seeing 6.5% longer runtime to instantiate an empty class with this decorator, which is a small impact on import time for pandas.

These are certainly small costs, so if there is a gain to be had then great. But I'm just not seeing what the gain is.

Reply:
To be honest, I don't think I had though about that option for classes (setting it afterwards like `DataFrame.__module__ = "pandas"` would move it further away, but is also an option), but we just copied the approach used in numpy.

If it has overhead, it's certainly another reason to do it the other way (and agree that too many decorators make the code harder too read)

Reply:
I assume that as the decorator is untyped, we also loose type definitions when applied to functions but I can add this to the todo list (along with a whats new) in the issue itself.

And also investigate the testing so that no public api classes/functions get missed.

Rather than holding up the merging of the open PRs, this discussion should probably be moved to the open issue also. 

I'll go ahead and get the open ones merged so that I can update the issue with what's left to close out the issue.

The task was more about checking whether any other changes were needed and it was only the Series one that needed other changes. So if we want to not use the decorator this could be an easy followup rather than a blocker.